consensus/common_test.go

```diff
@@ -326,6 +327,30 @@ func ensureNewStep(stepCh <-chan interface{}) {
 	}
 }
 
+func ensureVote(voteCh chan interface{}, height int64, round int, voteType byte) {
+	timer := time.NewTimer(ensureTimeout)
+	select {
+	case <-timer.C:
+		break
+	case v := <-voteCh:
+		edv, ok := v.(types.EventDataVote)
+		if !ok {
+			panic(fmt.Sprintf("expected a *types.Vote, got %v. wrong subscription channel?",
+				reflect.TypeOf(v)))
+		}
+		vote := edv.Vote
+		if vote.Height != height {
+			panic(fmt.Sprintf("expected height %v, got %v", height, vote.Height))
```

Comment:
Instead of panics this function should return errors, which makes it easier to report back in the tests.